lib/screens/services/add-service/add_service_form.dart

```diff
@@ -0,0 +1,134 @@
+
+  Future getImage() async {
+    final imageSource = await AppDialogs.chooseImageSource(context);
+
+    if (imageSource != null) {
+      final file = await ImagePicker.pickImage(source: imageSource);
+      if (file != null) {
+        setState(() => _image = file);
+      }
+    }
+  }
+
+  @override
+  void dispose() {
+    nameController.dispose();
+    categoryController.dispose();
+    descriptionController.dispose();
+    addressController.dispose();
+    petTypeController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Service service = ModalRoute
+        .of(context)
+        .settings
+        .arguments;
+
+    return Form(
+      key: _formKey,
+      child: Padding(
+        padding: EdgeInsets.symmetric(horizontal: 20.0),
+        child: Column(
+          children: <Widget>[
+            ProfilePicture(
+                image: _image,
+                pictureUrl: service.pictureUrl,
+                placeholderImageUri: "assets/blank_profile.png",
+                imageGetter: getImage),
+            InputField(
+              controller: nameController,
+              hintText: "Name",
+            ),
+            InputField(
+              controller: categoryController,
+              hintText: "Category e.g. 'Saloon', 'Clinic'",
+            ),
+            InputField(
+              controller: descriptionController,
+              hintText: "Description",
+            ),
+            InputField(
+              controller: addressController,
+              hintText: "Address",
+            ),
+            InputField(
+              controller: petTypeController,
+              hintText: "Animal Type e.g. 'Dog'",
+            ),
+            Padding(
+              padding: const EdgeInsets.symmetric(vertical: 15.0),
+              child: RaisedButton(
+                onPressed: () {
+                  if (_formKey.currentState.validate()) {
+                    addService(service);
+                  } else
+                    AppDialogs.showAlertDialog(context, "Operation failed",
+                        "Please, make sure that the inputs are in the correct format!");
+                },
+                child: Text('Add Service'),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void addService(Service serviceObject) {
+    serviceObject.ownerId = _authService.currentUserUid;
+    serviceObject.name = nameController.text;
+    serviceObject.category = categoryController.text;
+    serviceObject.description = descriptionController.text;
+    serviceObject.address = addressController.text;
+    serviceObject.petType = petTypeController.text;
+
+    widget.addServiceHandler(serviceObject);
+
+    if (_image != null) {
+      _storageService.uploadPhoto(_image).then((pictureUrl) {
+        serviceObject.pictureUrl = pictureUrl;
+
+        _serviceService
+            .updateService(serviceObject);
+      });
+    }
```

Comment:
Since `addServiceHandler` is async, it is possible for `updateService` to be called before the service was added.
Please fix this.